### PR TITLE
Do not set content type on empty requests

### PR DESCRIPTION
**Pull request by achamayou**
And add test for empty POSTs (described in #2191), which failed to spot any problems:

```
./tests.sh -VV -R governance_no_session_auth_test --repeat-until-fail 100
...
1/1 Test #56: governance_no_session_auth_test ...   Passed   26.78 sec

The following tests passed:
        governance_no_session_auth_test

100% tests passed, 0 tests failed out of 1

Label Time Summary:
cft    =  26.78 sec*proc (1 test)
e2e    =  26.78 sec*proc (1 test)

Total Test time (real) = 2830.34 sec
```

### CODE REVIEW

File: tests/governance.py [hunk 1]
```diff
@@ -229,6 +229,13 @@ def read_service_principal():
     return network
 
 
+@reqs.description("Test ack state digest updates")
+def test_ack_state_digest_update(network, args):
+    for _ in range(50):
```
Comment (wintersteiger) [inline tests/governance.py:234]: Seems like an arbitrary number - is it necessary to try this that often?

**Review by wintersteiger — APPROVED** [review]

File: tests/governance.py (hunk 1 above)
Comment (achamayou) [reply tests/governance.py:234]: It is arbitrary, but users have reported in #2191 that they sometimes see validation failures even when they have an empty body/content-length zero. Since there is as far as I can tell no way for that to happen, this is all I can think to do to increase our chances of catching this issue if it exists. Ideas welcome.

Comment (wintersteiger) [reply tests/governance.py]: I see, thanks for the explanation! The only thing I can think of is to add random (small) delays on various operations so we get unusual execution sequences. Not sure that's really a factor here though.

Comment (achamayou) [reply tests/governance.py]: Come to think of it what we should test is mismatched content length and sizes probably.
